docs/docs/recipes.md

```diff
@@ -770,6 +770,124 @@ export const pageQuery = graphql`
 - [Guide to creating pages from data programmatically](/docs/programmatically-create-pages-from-data/)
 - [Example repo](https://github.com/gatsbyjs/gatsby/tree/master/examples/recipe-sourcing-markdown) for this recipe
 
+### Sourcing data from Contentful
+
+#### Prerequisites
+
+- A [Contentful account](https://www.contentful.com/)
+- A Contentful space
+- One [Content model](https://www.contentful.com/developers/docs/concepts/data-model/)
+- One or more pieces of published content
+- The [space ID and content delivery API key](https://www.contentful.com/developers/docs/concepts/apis/#content-delivery-api) for your space
+
+For the screenshots and examples in this recipe, we have set up a new Contentful space with a "News" content model that includes "Title" and "Body" fields.
+
+**Note:** Per [this issue](https://github.com/gatsbyjs/gatsby/issues/15344#issuecomment-520151776), you should create and publish at least one piece of media in your Contentful space to avoid build errors.
+
+#### Directions
+
+##### Install and setup the Contentful plugin
+
+Install the `gatsby-source-contentful` plugin in your Gatsby site:
+
+```shell
+npm install --save gatsby-source-contentful
+```
+
+Add the following plugin entry to your `gatsby-config.js` file. Use your own space ID and access token from the API keys section of your Contentful space. You may also use [environment variables](https://www.gatsbyjs.org/docs/environment-variables/) to store your space and token.
+
+```javascript:title=gatsby-config.js
+plugins: [
+  {
+    resolve: `gatsby-source-contentful`,
+    options: {
+      spaceId: `your_space_id`,
```

Comment:
Should we add here directly the env to show good practice or it is out of scope here?

```suggestion
      spaceId: process.env.CONTENFUL_SPACE_ID,
```

Reply:
I went back and forth on that, I think by linking to the environment variable page we might be covered.

Reply:
I'm sorta torn by this, too. Does committing it as-is present a security risk?

Reply:
as I read from the docs there are readonly keys and also management access keys. If a newbie now expose this keys to a GitHub repo then it is not that good. Content can be manipulated or also (the user thinks it's) hidden fields can be read which are not for.public.

Reply:
env vars might make this recipe harder, what if we put in a disclaimer about not exposing it to the public. Maybe we need a page on gatsby/docs that talks about secrets.

Reply:
even the [Contentful README](https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby-source-contentful/README.md) shows this hint and a shortcut to https://gatsby.dev/env-vars

```
        spaceId: `your_space_id`,
        // Learn about environment variables: https://gatsby.dev/env-vars
        accessToken: process.env.CONTENTFUL_ACCESS_TOKEN,
```

Reply:
i think we make a receipt not harder, when we show new users directly at the beginning the only secure way.

Reply:
Maybe we should reword the sentence before the example to say "You should" instead "You may"?

Also, blushing a bit here, I included a space and access token so that people could just use it against a known space - which I think might be a bit flimsy.

Reply:
@wardpeet we have the https://gatsby.dev/env-vars page, but there is no section about security of private keys. good point.

Reply:
Works for me!

Reply:
Added in https://github.com/gatsbyjs/gatsby/pull/17318/commits/1eb7a30ca348992fc0223370e63084c3cc9466a9, should we consider this resolved for now? I agree that a general best practice for secure variables should be outlined in https://gatsby.dev/env-vars - should an issue be opened for that?

Reply:
That sounds like a great follow-up issue, so we can keep the scope of this PR focused.